src/gengowatcher/browser_session.py

```
@@ -0,0 +1,234 @@
+    accept_language: str = ""
+    origin: str = DEFAULT_GENGO_ORIGIN
+    target_url: str = ""
+    target_title: str = ""
+
+
+def _normalize_debug_url(debug_url: str | None) -> str:
+    return (debug_url or DEFAULT_BROWSER_DEBUG_URL).rstrip("/")
+
+
+def _load_cdp_targets(debug_url: str) -> list[dict[str, Any]]:
+    with urllib.request.urlopen(f"{debug_url}/json/list", timeout=5) as response:
```

Ruff (0.15.5)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)